packages/access-api/test/store-list.js

```diff
@@ -0,0 +1,197 @@
+      },
+    })
+    const spaceCreation = await createSpace(
+      issuer,
+      service,
+      conn,
+      'space-info@dag.house'
+    )
+    const listInvocation = Store.list.invoke({
+      issuer,
+      audience: service,
+      proofs: [spaceCreation.delegation],
+      with: spaceCreation.space.did(),
+      nb: {},
+    })
+    const result = await listInvocation.execute(
+      /** @type {import('@ucanto/interface').ConnectionView<any>} */ (conn)
+    )
+    if (result?.error) {
+      try {
+        /**
+         * This is expected to error even when the forwarding is working.
+         * When this happens, the error will be like
+         * InvalidAudience: Delegation audience is 'did:key:z6MkqBzPG7oNu7At8fktasQuS7QR7Tj7CujaijPMAgzdmAxD' instead of 'did:web:web3.storage'\n"
+         * This makes sense because this test is just ensuring that the invocation is forwarded based on the capability.can value
+         * even though the audience is just some random did:key
+         */
+        assert.ok('name' in result, 'result has name property')
+        assert.strictEqual(
+          result.name,
+          'InvalidAudience',
+          'error has expected name'
+        )
+        assert.strictEqual(
+          result.message?.includes(
+            `Delegation audience is '${service.did()}' instead of 'did:web:web3.storage'`
+          ),
+          true,
+          `error includes expected message`
+        )
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('unexpected error', result)
+        throw error
+      }
+    }
+  })
+  it('forwards invocations with aud=did:web:web3.storage', async function () {
+    const web3storageDid = 'did:web:web3.storage'
+    // if this is set, it's to inject in the actual private key used by web3StorageDid.
+    // and if it's present, the assertions will expect no error from the proxy or upstream
+    const privateKeyFromEnv = process.env.WEB3_STORAGE_PRIVATE_KEY
+    const {
+      issuer,
+      service: serviceSigner,
+      conn,
+    } = await context({
+      environment: {
+        ...process.env,
+        // this emulates the configuration for deployed environments,
+        // which will allow the access-api ucanto server to accept
+        // invocations where aud=web3storageDid
+        DID: web3storageDid,
+        PRIVATE_KEY: privateKeyFromEnv ?? process.env.PRIVATE_KEY,
+      },
+    })
+    const service = serviceSigner.withDID(web3storageDid)
+    const spaceCreation = await createSpace(
+      issuer,
+      service,
+      conn,
+      'space-info@dag.house'
+    )
+    const listInvocation = Store.list.invoke({
+      issuer,
+      audience: service,
+      proofs: [spaceCreation.delegation],
+      with: spaceCreation.space.did(),
+      nb: {},
+    })
+    const result = await listInvocation.execute(
+      // cast to `any` only because this `conn` uses Service type from access-client.
```

Comment:
Nit: We should probably change `access-client` as well so it's aware of `store` and ` upload` capabilities now. Can be a followup though.

Reply:
I thought about that but wasn't sure since I haven't hacked much in `access-client` yet. Let's do as followup as-needed

Reply:
https://github.com/web3-storage/w3protocol/issues/340